packages/datagateway-dataview/src/views/table/datafileTable.component.tsx

```diff
@@ -161,7 +161,7 @@ const DatafileTable = (props: DatafileTableProps): React.ReactElement => {
 
   return (
     <Table
-      loading={addToCartLoading || removeFromCartLoading}
+      loading={addToCartLoading || removeFromCartLoading || cartLoading}
```

Comment:
Is it worth adding an `allIdsLoading` as well? Since `allIds` is used in the `SelectHeader` onClick function and it can be undefined, this may lead to the `SelectHeader` being enabled when it doesn't have `allIds`

Reply:
allIds already has a fallback where it instead just uses the currently loaded `data` - although it's probably better to just disable it if allIds hasn't loaded. I'll add it to the loading stuff alongside `cartLoading`